packages/ag-charts-community/src/util/textMeasurer.ts

```diff
@@ -0,0 +1,332 @@
+    maxWidth: number;
+    maxLines?: number;
+    maxHeight?: number;
+    lineHeight?: number;
+    textWrap: TextWrap;
+}
+
+// Metrics for a single line of text.
+export interface LineMetrics {
+    width: number;
+    height: number;
+    offsetTop: number;
+    offsetLeft: number;
+    lineHeight: number;
+}
+
+// Metrics for multiple lines of text.
+export interface MultilineMetrics {
+    width: number;
+    height: number;
+    offsetTop: number;
+    offsetLeft: number;
+    lineMetrics: ({ text: string } & LineMetrics)[];
+}
+
+// TextMetrics adjusted for compatibility and extensibility.
+export interface LegacyTextMetrics extends Writeable<TextMetrics> {
+    emHeightAscent: number;
+    emHeightDescent: number;
+}
+
+// Manages text measurement and wrapping functionalities.
+export class TextMeasurer {
+    static readonly EllipsisChar = '\u2026'; // Representation for text clipping.
+
+    private static instanceMap = new Map<string, TextMeasurer>();
+    private static lineSplitter = /\r?\n/g;
+
+    // Creates or retrieves a TextMeasurer instance for a specific font.
+    private static createFontMeasurer(font: string) {
+        const ctx = createCanvasContext();
+        const measurer = new TextMeasurer(ctx);
+        TextMeasurer.instanceMap.set(font, measurer);
+        ctx.font = font;
+        return measurer;
+    }
+
+    // Gets a TextMeasurer instance, configuring text alignment and baseline if provided.
+    private static getFontMeasurer(options: MeasureOptions) {
+        const measurer = TextMeasurer.instanceMap.get(options.font) ?? TextMeasurer.createFontMeasurer(options.font);
+        if (options.textAlign) {
+            measurer.ctx.textAlign = options.textAlign;
+        }
+        if (options.textBaseline) {
+            measurer.ctx.textBaseline = options.textBaseline;
+        }
+        return measurer;
+    }
+
+    static wrapText(text: string, options: WrapOptions) {
+        return this.wrapLines(text, options).join('\n');
+    }
+
+    static wrapLines(text: string, options: WrapOptions) {
+        const lines: string[] = text.split(TextMeasurer.lineSplitter);
+        const measurer = TextMeasurer.getFontMeasurer(options);
+
+        if (options.textWrap === 'never') {
+            return lines.map((line) => this.truncateLine(line.trimEnd(), measurer, options.maxWidth));
+        }
+
+        const result: string[] = [];
+        const wrapHyphenate = options.textWrap === 'hyphenate';
+        const wrapOnSpace = options.textWrap === 'on-space';
+
+        for (let line of lines) {
+            line = line.trimEnd();
+
+            for (let i = 0, estimatedWidth = 0, lastSpaceIndex = 0; i < line.length; i++) {
+                const char = line.charAt(i);
+
+                estimatedWidth += measurer.textWidth(char);
+
+                if (char === ' ') {
+                    lastSpaceIndex = i;
+                }
+
+                if (estimatedWidth > options.maxWidth) {
```

Comment:
Nit: Would it improve readability if we decomposed this block a bit (maybe in a distinct function)?